tests/dataloader_tests.cpp

```diff
@@ -9,74 +9,75 @@
 
 /**
  * Simple test for Dataloader.
  */
-BOOST_AUTO_TEST_CASE(CSVDataLoaderTest)
+TEST_CASE("CSVDataLoaderTest", "[DataLoadersTest]")
 {
   // Check that dataloader loads only known datasets.
-  BOOST_REQUIRE_THROW(DataLoader<>("no-dataset", true), std::runtime_error);
+  REQUIRE_THROWS_AS(DataLoader<>("no-dataset", true), std::runtime_error);
 
   // Check Load CSV Function for Dataloader.
   Utils::DownloadFile("/datasets/iris.csv", "./../data/iris.csv");
+  
   // Check the file has been downloaded.
-  BOOST_REQUIRE(Utils::PathExists("./../data/iris.csv"));
+  REQUIRE(Utils::PathExists("./../data/iris.csv") == true);
 
   DataLoader<> irisDataloader;
   irisDataloader.LoadCSV("./../data/iris.csv", true, true, 0.5, false, false,
       0, -1, 1, -1);
 
   // Check for length and columns of training dataset.
-  BOOST_REQUIRE_EQUAL(irisDataloader.TrainLabels().n_cols, 75);
-  BOOST_REQUIRE_EQUAL(irisDataloader.TrainLabels().n_rows, 3);
+  REQUIRE(irisDataloader.TrainLabels().n_cols == 75);
+  REQUIRE(irisDataloader.TrainLabels().n_rows == 3);
 
   // Check for validation data as well.
-  BOOST_REQUIRE_EQUAL(irisDataloader.ValidFeatures().n_cols, 75);
-  BOOST_REQUIRE_EQUAL(irisDataloader.ValidFeatures().n_rows, 4);
+  REQUIRE(irisDataloader.ValidFeatures().n_cols == 75);
+  REQUIRE(irisDataloader.ValidFeatures().n_rows == 4);
 
   // Check for validation dataset using tuples.
-  BOOST_REQUIRE_EQUAL(std::get<1>(irisDataloader.ValidSet()).n_cols, 75);
-  BOOST_REQUIRE_EQUAL(std::get<1>(irisDataloader.ValidSet()).n_rows, 3);
+  REQUIRE(std::get<1>(irisDataloader.ValidSet()).n_cols == 75);
+  REQUIRE(std::get<1>(irisDataloader.ValidSet()).n_rows == 3);
 
   // Check for training dataset using tuples.
-  BOOST_REQUIRE_EQUAL(std::get<0>(irisDataloader.TrainSet()).n_cols, 75);
-  BOOST_REQUIRE_EQUAL(std::get<0>(irisDataloader.TrainSet()).n_rows, 4);
+  REQUIRE(std::get<0>(irisDataloader.TrainSet()).n_cols == 75);
+  REQUIRE(std::get<0>(irisDataloader.TrainSet()).n_rows == 4);
 
   Utils::RemoveFile("./../data/iris.csv");
 }
 
 /**
  * Simple test for MNIST Dataloader.
  */
-BOOST_AUTO_TEST_CASE(MNISTDataLoaderTest)
+TEST_CASE("MNISTDataLoaderTest", "[DataLoadersTest]")
 {
   DataLoader<> dataloader("mnist", true, 0.80);
 
   // Check for correct dimensions.
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_rows, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_rows, 784);
+  REQUIRE(dataloader.TrainFeatures().n_rows == 784);
+  REQUIRE(dataloader.TestFeatures().n_rows == 784);
+  REQUIRE(dataloader.ValidFeatures().n_rows == 784);
 
   // Check for correct dimensions.
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_cols, 8400);
-  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 33600);
-  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_cols, 28000);
+  REQUIRE(dataloader.TrainFeatures().n_cols == 8400);
+  REQUIRE(dataloader.ValidFeatures().n_cols == 33600);
```

Comment:
To solve the failing test, can you remove:

https://github.com/mlpack/models/blob/47ba8165a535e0f3f9d50481965f4e16653da19e/dataloader/dataloader_impl.hpp#L143-L144

With https://github.com/mlpack/mlpack/pull/2892 merged the header is already removed correctly.

Reply:
done.

Reply:
I would have never figured this out on my own that a change in mlpack was causing this.
Thank You so much.